twine/wininst.py

```diff
@@ -3,21 +3,16 @@
 import zipfile
 from typing import Optional
 
-from pkginfo import distribution
-
+from twine import distribution
 from twine import exceptions
 
 wininst_file_re = re.compile(r".*py(?P<pyver>\d+\.\d+)\.exe$")
 
 
 class WinInst(distribution.Distribution):
-    def __init__(self, filename: str, metadata_version: Optional[str] = None) -> None:
-        self.filename = filename
-        self.metadata_version = metadata_version
-        self.extractMetadata()
 
     @property
-    def py_version(self) -> str:
+    def py_version(self) -> Optional[str]:
```

Comment:
Everything at this level of `twine` is intended and in-scope for another project to import. Changing the signatures like this is thus a backwards incompatible change.

Are folks actually importing these? Probably not. But they could be and that's a risk I'm worried about. We'll need to make sure we release this as `7.0`.

Reply:
This is `Optional[str]` to keep the interface unified with all the implementations of `twine.distribution.Distribution`. The only subclass that returns `None` is however `twine.sdist.SDist` and it should rather return `"source"` as indicated in the documentation for the package index submission format. However, I didn't wan to add another cleanup on top. I can submit another PR to do that and then restore this to return `str`.

Reply:
I restored the signature. See also #1191.